src/ast.c

```diff
@@ -375,6 +376,7 @@ void jl_init_common_symbols(void)
     jl_release_sym = jl_symbol("release");
     jl_acquire_release_sym = jl_symbol("acquire_release");
     jl_sequentially_consistent_sym = jl_symbol("sequentially_consistent");
+    jl_julia_char_sym = jl_symbol("julia_char");
```

Comment:
It's a small thing but we can avoid the need for this symbol, since this is never used as a julia-level expression head. Instead we can use `fl_ctx->jl_char_sym` prior to calling `scmsym_to_julia` to check for this expression.